Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/HDInsightAdditionalClusterDetail.java

```diff
@@ -139,36 +104,21 @@ public int getDataNodes() {
     }
 
     @Override
-    public String getHttpUserName() throws HDIException {
+    public String getHttpUserName() {
```

Comment:
`@Nullable`?

Reply:
For linked cluster, user name and password are necessary. Will make it @NotNull

---
In reply to: [274297437](https://github.com/Microsoft/azure-tools-for-java/pull/3001#discussion_r274297437) [](ancestors = 274297437)